staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -244,17 +213,25 @@ func (wc *watchChan) startWatching(watchClosedCh chan struct{}) {
 	}
 
 	klog.V(3).Infof("Starting watcher for wc.ctx=%v, wc.key=%v", wc.ctx, wc.key)
-	m := parseConfig(wc.partitionConfigFilename)
 
-	if val, ok := m[wc.key]; ok {
-		rangeStart := wc.key + val[0]
-		rangeEnd := wc.key + val[1]
-
-		wc.key = rangeStart
-		opts = append(opts, clientv3.WithRange(rangeEnd))
-		klog.V(3).Infof("Assigned watcher to partition wc.key=%v, endKey=%v", wc.key, rangeEnd)
+	if val, ok := wc.partitionConfig[wc.key]; ok {
```

Comment:
Using array for range makes the logic harder to read & maintain. Please use struct instead. When parsing the configuration file, assign range to different field such as start, end. You may use flags for open end ranges.